ddtrace/_trace/processor/__init__.py

```diff
@@ -304,7 +304,22 @@ def on_span_finish(self, span):
         # type: (Span) -> None
         with self._lock:
             self._span_metrics["spans_finished"][span._span_api] += 1
+
+            if span.trace_id not in self._traces:
+                log_msg = f"No trace found for span {span} (was tracer reconfigured?)"
+                if config._telemetry_enabled:
+                    telemetry.telemetry_writer.add_log("WARNING", log_msg)
+                log.warning(log_msg)
+                return
+
             trace = self._traces[span.trace_id]
+            if span not in trace.spans:
```

Comment:
I wonder if this is one of the contributors to the execution time increase on the tracer large benchmark?